tapfreighter/chain_porter.go

```diff
@@ -870,11 +870,45 @@ func (p *ChainPorter) transferReceiverProof(pkg *sendPackage) error {
 
 	// If we have a non-interactive proof, then we'll launch several
 	// goroutines to deliver the proof(s) to the receiver(s).
-	err := fn.ParSlice(ctx, pkg.OutboundPkg.Outputs, deliver)
+	instanceErrors, err := fn.ParSliceErrCollect(
+		ctx, pkg.OutboundPkg.Outputs, deliver,
+	)
 	if err != nil {
 		return fmt.Errorf("error delivering proof(s): %w", err)
 	}
 
+	// If there were any errors during the proof delivery process, we'll
+	// report them here and return the first error we encountered.
+	if len(instanceErrors) > 0 {
+		var firstErr error
+		for idx := range instanceErrors {
+			output := pkg.OutboundPkg.Outputs[idx]
+			instanceErr := instanceErrors[idx]
+
+			// Store the first error encountered.
+			if firstErr == nil {
+				firstErr = instanceErr
+			}
+
+			scriptPubKey :=
+				output.ScriptKey.PubKey.SerializeCompressed()
+			anchorOutpoint := output.Anchor.OutPoint.String()
+			courierAddr := string(output.ProofCourierAddr)
+
+			log.Errorf("Error delivering transfer output proof "+
+				"(anchor_outpoint=%s, script_pub_key=%v, "+
+				"position=%d, proof_courier_addr=%s, "+
+				"proof_delivery_status=%v): %v",
+				anchorOutpoint, scriptPubKey, output.Position,
+				courierAddr, output.ProofDeliveryComplete,
+				instanceErr)
+		}
+
+		if firstErr != nil {
```

Comment:
Instead of tracking this separately, I think we could add a function `fn.PeekMap()` that returns the first element of a map?

Reply:
Also, shouldn't `firstErr` always be non-nil at this point, as the map of errs is non-empty and there should be no entry will a nil value?

Reply:
> Also, shouldn't firstErr always be non-nil at this point, as the map of errs is non-empty and there should be no entry will a nil value?

I've removed `if len(instanceErrors) > 0 {` in the latest commit, you were right @jharveyb that some of the code was redundant in the previous version.